ui/brave_custom_notification/message_popup_view.cc

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ui/brave_custom_notification/message_popup_view.h"
+#include "brave/ui/brave_custom_notification/message_view.h"
+#include "brave/ui/brave_custom_notification/message_view_factory.h"
+#include "brave/ui/brave_custom_notification/public/cpp/constants.h"
+#include "build/build_config.h"
+#include "ui/display/display.h"
+#include "ui/display/screen.h"
+#include "ui/views/layout/fill_layout.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/controls/textfield/textfield.h"
+#include "ui/views/controls/label.h"
+
+#if defined(OS_WIN)
+#include "ui/views/widget/desktop_aura/desktop_native_widget_aura.h"
+#endif
+
+#if defined(OS_CHROMEOS)
+#include "ui/aura/window.h"
+#include "ui/aura/window_targeter.h"
+#endif
+
+namespace brave_custom_notification {
+namespace {
+static std::map<std::string, MessagePopupView*> g_notifications_;
+}
+
+// static
+void MessagePopupView::Show(const Notification& notification) {
+  g_notifications_[notification.id()] = new MessagePopupView(notification);
+}
+
+// static
+void MessagePopupView::Clicked(const std::string& notification_id) {
+  MessagePopupView* message_popup_view = g_notifications_[notification_id];
+  message_popup_view->Close();
+  message_popup_view->notification_.delegate()->Click(base::nullopt, base::nullopt);
+}
+
+// static
+void MessagePopupView::ClosePopup() {
+  for (auto iter = g_notifications_.begin(); iter != g_notifications_.end(); ++iter) {
+    MessagePopupView* message_popup_view = g_notifications_[iter->first];
+    message_popup_view->Close();
+    message_popup_view->notification_.delegate()->Close(true);
+  }
+}
+
+MessagePopupView::MessagePopupView(const Notification& notification) :
+  notification_(notification) {
+  SetLayoutManager(std::make_unique<views::FillLayout>());
+  views::Widget::InitParams params(views::Widget::InitParams::TYPE_POPUP);
+  params.type = views::Widget::InitParams::TYPE_WINDOW_FRAMELESS;
+  params.z_order = ui::ZOrderLevel::kFloatingWindow;
+  params.bounds = { 30, 30, 300, 100 + GetBodyHeight(notification.message())};
```

Comment:
Can we please move hardcoded values to `const` inside an empty namespace in the `.cc` file